core/src/main/java/software/amazon/awssdk/async/FileAsyncResponseHandler.java

```diff
@@ -66,13 +66,8 @@ public void onStream(Publisher<ByteBuffer> publisher) {
     public void exceptionOccurred(Throwable throwable) {
         try {
             invokeSafely(fileChannel::close);
-        } catch (RuntimeException e) {
-            path.toFile().delete();
-            throw e;
-        }
-        if (!path.toFile().delete()) {
-            throw new UncheckedIOException(new IOException(
-                    String.format("Could not delete %s.", path.toFile().getAbsolutePath())));
+        } finally {
+            invokeSafely(() -> Files.delete(path));
```

Comment:
Won't this hide the original exception if we fail to delete the file?

Reply:
Yes, but "I couldn't delete this file that may still sitting on your disk somewhere" seemed more important than the fact that we couldn't close the file channel anyway, so I went ahead and made the behavior change. Let me know if you don't think that's appropriate.

Reply:
Actually nvm, I believe we don't care if an exception is thrown from this callback and we complete the future with the original exception that cause this to be invoked which sounds like the right behavior.

https://github.com/aws/aws-sdk-java-v2/blob/master/core/src/main/java/software/amazon/awssdk/http/pipeline/stages/MakeAsyncHttpRequestStage.java#L183